### PR TITLE
Add snippets for swift

### PR DESCRIPTION
**Pull request by jojojames**

### CODE REVIEW

File: snippets/swift-mode/uiviewcontrollerlifecycle [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# -*- mode: snippet -*-
+# name: uiViewControllerLifecycle
+# key: uiViewControllerLifecycle
+# --
+// MARK: UIViewController lifecycle
```
Comment (AndreaCrotti) [inline snippets/swift-mode/uiviewcontrollerlifecycle]: this one seems quite long, do you always have to write all this code together as a single unit?

File: snippets/swift-mode/uitableviewdelegate [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# -*- mode: snippet -*-
+# name: uiTableViewDelegate
+# key: uitableViewDelegate
+# --
+// MARK: UITableViewDelegate
```
Comment (AndreaCrotti) [inline snippets/swift-mode/uitableviewdelegate]: same here, also there is nothing to expand here, is this really just boilerplate code that everyone has to copy & paste around (sorry I know nothing about Swift & co)

**Comment by AndreaCrotti** [comment]
Thanks a lot @jojojames 
Some snippets seem more framework specific other than actual Swift only (forgive my ignorance).
Are they really all generic enough that most Swift developers would use these snippets?

File: snippets/swift-mode/uiviewcontrollerlifecycle (hunk 1 above)
Comment (jojojames) [reply snippets/swift-mode/uiviewcontrollerlifecycle]: Yeah. It's very common to pull those methods in and ViewControllers are very common in iOS land. (One of the big 3 (M.V.Controllers)

**Comment by jojojames** [comment]
Yup! Most of them take inspiration from https://github.com/burczyk/XcodeSwiftSnippets so there's some precedence in them being useful. :)

**Comment by AndreaCrotti** [comment]
Ok makes sense

**Pull request merged**